src/cpr_data_access/models.py

```diff
@@ -1025,6 +1026,18 @@ def filter_by_language(self, language: str) -> "Dataset":
         """Return documents whose only language is the given language."""
         return self.filter("languages", [language])
 
+    def sample(self, n: int, random_state: int = 42) -> "Dataset":
+        """Samples n documents from the dataset and returns a Dataset object with only those."""
+        random.seed(random_state)
+
+        documents = random.sample(self.documents, n)
+
+        instance_attributes = {
+            k: v for k, v in self.__dict__.items() if k != "documents"
```

Comment:
Yes, good point -- in all honesty, I've copied this from the `filter()` implementation. But I do agree that a `.dict()` method is needed, let me add that.

Reply:
On the attribute selection -- we'd want all the attributes, because the returned `Dataset` should have the same attributes, except fewer documents, as those have been sampled. 

I think this also insulates us from having to change the .dict() method every time an attribute is added to / removed from the `Dataset` object.

Reply:
Okay nice one! Happy days